util/collect_metrics_files.rb

```diff
@@ -0,0 +1,253 @@
+# frozen_string_literal: true
+
+require "optparse"
+require "time"
+require "tmpdir"
+
+END_TIME_DEFAULT = Time.now.to_i
+START_TIME_DEFAULT = END_TIME_DEFAULT - (8 * 60 * 60) # minus 8 hours
+# TODO: can we get this from the module, it is configurable
+METRICS_DIRECTORY_DEFAULT = "/opt/puppetlabs/puppet-metrics-collector"
+# TODO: can we get this from the module, it is configurable
+POLL_INTERVAL_DEFAULT = 60 * 5
+# archival runs at a random time between 12am and 3am.
+# But the random time should be static for a given service
+# TODO: can we get this from the module, it is configurable
+ARCHIVE_INTERVAL_DEFAULT = 60 * 60 * (24 + 0)
+OUTPUT_DIR_DEFAULT = Dir.mktmpdir
+
+# Built the default tar file name from the start and end times
+#
+# @author Randell Pelak
+#
+# @param [string] start_epoch epoch time for the start of the window
+# @param [string] end_epoch epoch time for the end of the window
+#
+# @return [string] Filename
+#
+# @example
+#   filename = build_default_tar_filename(start_epoch, end_epoch)
+def build_default_tar_filename(start_epoch, end_epoch)
+  start_time_str = Time.at(start_epoch).strftime("%Y%m%dT%H%M%SZ")
+  end_time_str = Time.at(end_epoch).strftime("%Y%m%dT%H%M%SZ")
+  return "#{start_time_str}-#{end_time_str}.tar.gz"
+end
+
+TAR_FILE_NAME_DEFAULT = build_default_tar_filename(START_TIME_DEFAULT, END_TIME_DEFAULT)
+
+DESCRIPTION = <<~DESCRIPTION
+  This script is designed to gather up the tar and json files from the metrics_collector module
+  within a specified time frame.  Going as far as opening up a tar file and extracting
+  only the matching jsons.
+DESCRIPTION
+
+DEFAULTS = <<~DEFAULTS
+  The following defaults values are used if the options are not specified:
+    * start (-s, --start): Now minus 8 hours. Ex: #{START_TIME_DEFAULT}
+    * end (-e, --end): Now. Ex: #{END_TIME_DEFAULT}
+    * metrics_dir (-t, --metrics_dir): #{METRICS_DIRECTORY_DEFAULT}
+    * output_dir (-t, --output_dir): Random temp dir.
+        Ex: #{OUTPUT_DIR_DEFAULT}
+    * tar_file_name (-t, --tar_file_name): Built from start and end time. Ex: #{TAR_FILE_NAME_DEFAULT}
+    * poll_interval (-p, --poll_interval): #{POLL_INTERVAL_DEFAULT}
+    * archive_interval (-a, --archive_interval): #{ARCHIVE_INTERVAL_DEFAULT}
+    * verbose (-v, --verbose): False
+DEFAULTS
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.banner = "Usage: collect_metrics_files.rb [options]"
+
+  opts.on("-h", "--help", "Display the help text") do
+    puts DESCRIPTION
+    puts opts
+    puts DEFAULTS
+    exit
+  end
+
```

Comment:
It would be helpful to have the option to collect all of the available data. Using `--start 0` seems to work but results in a name like '19700101T000000Z-20190805T192726Z' rather than the actual start of the data.

It would also be helpful to specify --start and --end with the format used in the default name so that a user could specify a date / time range without converting to unix format.

Reply:
Filed SLV-539 on this.